exporter/zipkinexporter/zipkin_test.go

```diff
@@ -339,3 +343,60 @@ const zipkinSpansJSONJavaLibrary = `
   "duration": 207000
 }]
 `
+
+// The rest of the fields should match up exactly
+func TestZipkinExporter_roundtripProto(t *testing.T) {
+	buf := new(bytes.Buffer)
+	var contentType string
+	cst := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		io.Copy(buf, r.Body)
+		contentType = r.Header.Get("Content-Type")
+		r.Body.Close()
+	}))
+	defer cst.Close()
+
+	tes, err := newZipkinExporter(cst.URL, "", time.Millisecond, true)
+	if err != nil {
+		t.Fatalf("Failed to create a new Zipkin receiver: %v", err)
+	}
+
+	// The test requires the spans from zipkinSpansJSONJavaLibrary to be sent in a single batch, use
+	// a mock to ensure that this happens as intended.
+	mzr := newMockZipkinReporter(cst.URL)
+	tes.reporter = mzr
+
+	mzr.serializer = zipkinproto.SpanSerializer{}
+
+	// Run the Zipkin receiver to "receive spans upload from a client application"
+	zexp := processor.NewTraceFanOutConnector([]consumer.TraceConsumer{tes})
+	zi, err := zipkinreceiver.New(":0", zexp)
```

Comment:
nit: use `testutils.GetAvailableLocalAddress(t)` instead of ":0" - in practice will use "localhost:0" avoiding a pop up on some lockdown systems, but also is consistent with other tests.